### PR TITLE
fix issue where we are not returning all the available ingesters in the ring

### PR DESCRIPTION
**Pull request by roystchiang**
**What this PR does**:

Fix an issue where extended write does not return all the available instances.

If we checked ingester-1 that's in `az-A`, and it's not active. We'll add `az-A` to distinct zones in the subsequent loops, we'll never check ingesters in  `az-A` again.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by alvinlin123** [comment]
With this change, we should see less CPU usage during deployment too.

File: pkg/ring/ring_test.go [hunk 1]
```diff
@@ -132,6 +132,62 @@ func TestAddIngesterReplacesExistingTokens(t *testing.T) {
 	require.Equal(t, newTokens, r.Ingesters[ing1Name].Tokens)
 }
 
+func TestRing_Get_ZoneAwarenessWithIngesterLeaving(t *testing.T) {
```
Comment (pracucci) [inline pkg/ring/ring_test.go:135]: Could you also add a test to cover the case you're running a number of zones > replication factor, please? I would like to make sure it behaves correctly too.

**Review by pracucci — COMMENTED** [review]
Thanks for spotting this issue! It's definitely an issue but I'm unsure about the fix. Please take a look at my comment.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Comment by pracucci** [comment]
@roystchiang Please make sure to sign your commits ([see here](https://cortexmetrics.io/docs/contributing/#developer-certificates-of-origin-dcos)) otherwise we can't merge.

**Comment by roystchiang** [comment]
> @roystchiang Please make sure to sign your commits ([see here](https://cortexmetrics.io/docs/contributing/#developer-certificates-of-origin-dcos)) otherwise we can't merge.

oops. Sorry. I will sign.

**Comment by pracucci** [comment]
> > @roystchiang Please make sure to sign your commits ([see here](https://cortexmetrics.io/docs/contributing/#developer-certificates-of-origin-dcos)) otherwise we can't merge.
> 
> oops. Sorry. I will sign.

Sorry to bother you. Every single commit have to be signed (not just the latest one). Either you squash to the signed commit or you sign every commit.

**Review by pstibrany — APPROVED** [review]
LGTM, thank you.

**Pull request merged**